### ISSUE TITLE
Instantiating Piecewise((a, Eq(x, y)), (b, True)) is extremely slow

### ISSUE THREAD

**Issue by rlamy** [sympy/sympy#11757]
Another issue (after #11717) found by looking at the [dsolve() benchmark](https://github.com/sympy/sympy_benchmarks/blob/master/benchmarks/dsolve.py): constructing Piecewise instances of the form `Piecewise((a, Eq(x, y)), (b, True))` takes way too long.

To demonstrate the problem, here's [a simple benchmark](https://gist.github.com/rlamy/810ee71df6a6b2d426e63386637742b6). I get timings of ~45 ms per instance creation on CPython 2.7, which is orders of magnitude more than the creation of such a simple object should take. 

[Profiling this under vmprof](http://vmprof.com/#/0fb1992184c7df9d46697a2c35307e12) shows that the vast majority of the time is spent in `nsimplify()` which is called by `random_complex_number()`.

There are actually two issues here:
- `random_complex_number(..., rational=True)` shouldn't call `nsimplify()`
- The constructor for Piecewise should not be attempting such complicated calculations in the first place.

**Comment by asmeurer** [comment]
Should this be closed?

**Comment by rlamy** [comment]
Well, the Piecewise constructor is still doing too much, and `random_complex_number(..., rational=True)` is still very slow, but it doesn't seem to be major bottleneck any more, so yes, probably.
